pkg/app/src/@types/app.ts

```diff
@@ -0,0 +1,35 @@
+import configJSON from 'src/data/config.json'
+import featuresJSON from 'src/data/features.json'
+
+export type AppConfigTypes = typeof configJSON.global
+export type AppFeatureTypes = typeof featuresJSON.global
+
+export interface AppAction {
+	type: any
+	payload?: object | null
+}
+
+export enum AppActionTypes {
+	CONFIG_LOAD = 'CONFIG_LOAD',
+	CONFIG_UPDATE = 'CONFIG_UPDATE',
+	FEATURES_LOAD = 'FEATURES_LOAD',
+	FEATURES_UPDATE = 'FEATURES_UPDATE',
+}
+
+export interface AppState {
```

Comment:
why interface for a struct vs types? interfaces make only sense if we represent the data structures as classes (e.g. for the gql) — but this opinion could be outdated as well.

Reply:
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces

Is basically the same the advantage of interface is that you can extend it

Reply:
yes, and types can be composed as well :)